client/utils/utils.test.js

```diff
@@ -143,14 +143,22 @@ describe("shouldLinkBeShown tests", () => {
     );
     expect(shouldLinkBeShown(link, isAuthenticated, userData)).toBe(false);
   });
-  it("should return false if user is authenticated and link.method is not mobile_phone", () => {
+  it("should return false if user is authenticated and link.method is not equal to userData.method", () => {
     const {link, isAuthenticated, userData} = createArgs(
-      {authenticated: true, method: "online_payment"},
+      {authenticated: true, method: "payment"},
       true,
-      {},
+      {method: "mobile_phone"},
     );
     expect(shouldLinkBeShown(link, isAuthenticated, userData)).toBe(false);
   });
+  it("should return true if userData.method and link.method is same", () => {
+    const {link, isAuthenticated, userData} = createArgs(
+      {authenticated: true, method: "payment"},
+      true,
+      {method: "payment"},
+    );
+    expect(shouldLinkBeShown(link, isAuthenticated, userData)).toBe(true);
+  });
```

Comment:
This is the failing test if we remove the changes made in commit https://github.com/openwisp/openwisp-wifi-login-pages/pull/269/commits/5d5161a3434aa620ece014144efdc9da4ded46b5